net/listener_grpc.go

```diff
@@ -197,3 +198,32 @@ func grpcHandlerFunc(grpcServer *grpc.Server, otherHandler http.Handler) http.Ha
 		}
 	})
 }
+
+//ControlListener is used to keep state of the connections of our drand instance
+type ControlListener struct {
+	conns *grpc.Server
+	lis   net.Listener
+}
+
+//NewTCPGrpcControlListener registers the pairing between a ControlServer and a grpx server
+func NewTCPGrpcControlListener(s control.ControlServer, port string) ControlListener {
+	lis, err := net.Listen("tcp", fmt.Sprintf("%s:%s", "localhost", port))
+	if err != nil {
+		slog.Fatal("Failed to listen")
+		return ControlListener{}
+	}
+	grpcServer := grpc.NewServer()
+	control.RegisterControlServer(grpcServer, s)
+	return ControlListener{conns: grpcServer, lis: lis}
+}
+
+func (g *ControlListener) Start() {
+	if err := g.conns.Serve(g.lis); err != nil {
+		slog.Fatalf("failed to serve: %s", err)
+	}
+}
+
+func (g *ControlListener) Stop() {
+	g.conns.Stop()
+	g.lis.Close()
```

Comment:
Juste closing the `grpc.Server` should  be enough (source: https://godoc.org/google.golang.org/grpc#Server.Stop )